### DISCUSSION TITLE
Querying the full contents of a specific webpage

### DISCUSSION THREAD

**Discussion by qunash** [qunash/chatgpt-advanced#52] [Announcements]
Hey everyone, 

Following a popular request, I've added a new beta feature – the ability to query the contents of a specific webpage.

To use the feature, simply include the `page:{URL}` operator in your query.
For example, `Extract key points from the article. page:https://myblog.com/rickrolling` will return the full text of the webpage at the provided URL.

If you'd like to try it out, you can install the 2.1.0 beta release manually: https://github.com/qunash/chatgpt-advanced#manual-installation.
(:exclamation:Don't forget to temporarily disable the webchatgpt extension you downloaded from the web store, at `chrome://extensions`)

I'd love to hear your feedback on the new feature!

**Comment by Norgus** [comment]
Is this an actual feature or a prank? (considering the rickroll reference and that the responses I get don't seem to touch on the full content of given pages)

**Reply by qunash** [reply]
It is an actual feature. It is experimental though, not all pages can be scraped correctly.
Which pages are you having issues with?

**Comment by Norgus** [comment]
In that case, sorry for the suggestion it might be a prank.

I remember that democracynow was the first one I tried, I don't know why it results in a 403 error, as the page will load in a tab fine. 
```[1] "Error extracting text from URL: Article `download()` failed with 403 Client Error: Forbidden for url: https://democracynow.org/ on URL https://democracynow.org"
URL: ```

I can't remember the other sites I tried and failed with, but I just got it to work with bbc news, although the loaded text is a bit too short to really have been worth doing.

```Web search results:

[1] "Toronto mayor quits over affair with ex-staffer

John Tory said the relationship was "a serious error in judgement" after it was revealed by a newspaper."
URL: https://www.bbc.co.uk/news


Current date: 11/02/2023

Instructions: Using the provided web search results, write a comprehensive reply to the given query. Make sure to cite results using [[number](URL)] notation after the reference. If the provided search results refer to multiple subjects with the same name, write separate answers for each subject.
Query: page:https://www.bbc.co.uk/news what's going on?```

**Reply by Norgus** [reply]
excuse the self-reply, but I just used page: on a specific bbc news article link and that brings in all the relevant text, so perhaps I just need to use it more that way.

**Reply by qunash** [reply]
Yes, it works better with article-style pages

**Reply by Norgus** [reply]
It seems like the BBC news page is quite inconsistent, my second and third article links didn't really load much in (one seemed to be prompting to load the mobile page, the other only loaded an article title)

**Reply by qunash** [reply]
Thanks for the feedback, I'll look into it.

**Comment by ramccall0** [comment]
Hi there! Great plugin! Would we be able to get multiple pages in one query? For example have us tell it to query multiple specific pages we want indexed?

**Comment by 6DiegoDiego9** [comment]
how do you overcome the 4K tokens limitation of ChatGPT input prompt?
